### ISSUE TITLE
Continuing configuration evolution when run budget is exhausted.

### ISSUE THREAD

**Issue by Bronzila** [automl/DEHB#67]
> Note: This issues is only relevant when the Config ID changes have been merged.

While waiting to fetch all results when the run budget is exhausted, we keep sampling/evolving new configurations:
```python
if self.is_worker_available():
    job_info = self._get_next_job()
    if brackets is not None and job_info["bracket_id"] >= brackets:
        # ignore submission and only collect results
```
While this was not an issue before, we would end up filling the `ConfigRepository` with unnecessary configurations, which will never be evaluated. We should think about how to redesign this.

**Comment by Bronzila** [comment]
This will most certainly be cleared up by merging the restarting PR, since there we adjust the way to check for collecting results.

**Comment by Bronzila** [comment]
Fixed in #81
